assignments/prototypes.js

```diff
@@ -102,9 +135,71 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+  function Villain(attr) {
+    Humanoid.call(this, attr);
+  };
+
+  Villain.prototype = Object.create(Humanoid.prototype);
+
+  Villain.prototype.attack = function(target) {
+    target.healthPoints -= 5;
+    if (target.healthPoints > 0) {
+      return `${this.name} attacks ${target.name} \n ${target.takeDamage()}`;
+    }
+    else {
+      return `${this.name} attacks ${target.name} \n ${target.destroy()}`;
+    }
+  };
+
+  function Hero(attr) {
+    Humanoid.call(this, attr);
+  };
+  
+  Hero.prototype = Object.create(Villain.prototype);
+
+
+  const evilWizard = new Villain({
+    createdAt: new Date(),
+    dimensions: {
+      length: 1,
+      width: 2,
+      height: 4,
+    },
+    healthPoints: 15,
+    name: 'Ganon',
+    team: 'Hyrule',
+    weapons: [
+      'Magic',
+      'Big fists',
+    ],
+    language: 'N64 nonsense noises',
+  });
+
+  const coolGuy = new Hero({
+    createdAt: new Date(),
+    dimensions: {
+      length: 1,
+      width: 2,
+      height: 4,
+    },
+    healthPoints: 10,
+    name: 'Cool Guy',
+    team: 'Weird Internet Videos',
+    weapons: [
+      'Sick Beats',
+      'Dance Moves',
+    ],
+    language: 'England',
+  });
+
+  console.log(evilWizard.attack(coolGuy));
+  console.log(coolGuy.attack(evilWizard));
+  console.log(coolGuy.attack(evilWizard));
+  console.log(evilWizard.attack(coolGuy));
```

Comment:
*team: 'Weird Internet Videos'* lol This is great, Bobby. Awesome job with the stretch.